packages/back-end/src/api/saved-groups/putSavedGroup.ts

```diff
@@ -0,0 +1,35 @@
+import { PutSavedGroupResponse } from "../../../types/openapi";
+import {
+  getSavedGroupById,
+  parseSavedGroupString,
+  toSavedGroupApiInterface,
+  updateSavedGroup,
+} from "../../models/SavedGroupModel";
+import { createApiRequestHandler } from "../../util/handler";
+import { putSavedGroupValidator } from "../../validators/openapi";
+
+export const putSavedGroup = createApiRequestHandler(putSavedGroupValidator)(
+  async (req): Promise<PutSavedGroupResponse> => {
+    const { groupName, groupList, owner } = req.body;
+    const { id } = req.params;
+
+    const savedGroup = await getSavedGroupById(id, req.organization.id);
+
+    if (!savedGroup) {
+      throw new Error(`Unable to locate the saved-group: ${id}`);
+    }
+
+    const updatedSavedGroup = await updateSavedGroup(id, req.organization.id, {
+      values: groupList ? parseSavedGroupString(groupList) : savedGroup.values,
+      groupName: groupName ? groupName : savedGroup?.groupName,
+      owner: owner ? owner : savedGroup?.owner,
+    });
```

Comment:
So far in the private API we have not been using the correct request methods for the actions we're performing, e.g. some of our PUT requests are more like PATCH requests, by definition. You can learn the difference [here](https://rapidapi.com/blog/put-vs-patch) but essentially puts should replace the entire resource (and should likely have the same validation logic as posts) whereas patches allow partial updates and all fields can be optional.

I'm wondering now that since we are doing a public API, if it's time to make that change to something developers are expecting. This implementation is a patch.

Also it looks like there's an issue with the `owner` field—if `owner` is not a required field, it looks like the user can't clear it by this logic since it'll fall back to the initial one if it's an empty string.

So what I'd probably do:

- Have PUT replace the full object, as is expected by developers implementing the API
- Create a PATCH method for patching only select fields

I don't think this approach of falling back to the initial resource's values is appropriate since it's not standard put behaviour, and also it means users cannot clear optional fields using this method.

Reply:
@tinahollygb Thanks for your feedback and for linking to that resource. I've updated the PUT request so its more like the POST in that it requires all required fields, and I've added a PATCH method that allows the user to pass in as few as 1 property they'd like to update.

Additionally, I have updated the logic so a user is capable of clearing the `owner` field if they pass in an empty string.